packages/world/src/systems/libraries/MoveLib.sol

```diff
@@ -79,17 +115,16 @@ library MoveLib {
     EntityId[] memory playerEntityIds = _removePlayerPosition(playerCoord);
     EntityId player = playerEntityIds[0];
 
-    (Vec3 finalCoord, uint128 totalCost) = _computeGravityResult(playerCoord, 0);
-
-    _setPlayerPosition(playerEntityIds, finalCoord);
+    MoveContext memory ctx = _moveContext(player);
 
-    uint128 currentEnergy = updatePlayerEnergy(player).energy;
```

Comment:
In move and jumpBuild we can just call getEnergy because we already updated the energy in the movesystem, but when gravity runs the player that's falling might not be the same player as the caller, and will not have their energy updated

Also hmm curious why any of the gravity tests didnt catch this

Reply:
yes I think the tests don't catch it because in most tests the block timestamp is the same throughout the test. I need to figure out how to fit the updatePlayerEnergy within the contract size limit 🫠

Reply:
did it by moving the responsibility of calling updatePlayerEnergy to the caller system

Reply:
hmm, but where is it still called for the gravity case?

Reply:
Like this only fixes the player you're mining, but `runGravity` is recursively called:

```
updatePlayerEnergy(above);
MoveLib.runGravity(aboveCoord);
```

Reply:
why wont it fit in `runGravity`? It's already a public function no?

Reply:
ahh it seems that with the latest refactor, MoveLib decreased in size just enough to fit that, so will add it to the moveContext function

Reply:
nicee! let's remove the `updateEnergy` from `Minelib._removeBlock` now then